x-pack/plugins/security_solution/common/endpoint/generate_data.ts

```diff
@@ -452,12 +455,36 @@ export class EndpointDocGenerator {
    * @param options - Allows event field values to be specified
    */
   public generateEvent(options: EventOptions = {}): EndpointEvent {
+    const processName = options.processName ? options.processName : randomProcessName();
+    const detailRecordForEventType =
```

Comment:
When reading `genreateEvent`, it may not always be necessary to understand the logic of calculating the detail record. 

Also, by defining this function and calling this function in a logical operator (`||`), the readability is hindered. It is unclear why `detailRecordForEvent` is replaced with `options.extensions` if `options.extensions` is truthy.

Also, `options.eventCategory` is rebound as `eventCategory`, which seems like an unnecessary indirection.

For these reasons, consider moving this logic into a private method:

```
private detailRecordForEventCategory(eventCategory: EventOptions['eventCategory']) {
// ...
}
```

Reply:
discussed, this is for testing

Reply:
tests got found